src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -434,10 +434,13 @@ private static void checkPreconditions(Project project) {
             throw new GradleException("Must be applied only to root project");
         }
 
-        Preconditions.checkState(
-                !project.getGradle().getStartParameter().isConfigureOnDemand(),
-                "Gradle Consistent Versions doesn't currently work with configure-on-demand, please remove"
-                        + " 'org.gradle.configureondemand' from your gradle.properties");
+        // check that configure-on-demand is false when running with --write-locks
+        if (project.getGradle().getStartParameter().isWriteDependencyLocks()) {
+            Preconditions.checkState(
+                    !project.getGradle().getStartParameter().isConfigureOnDemand(),
+                    "configure-on-demand cannot be enabled when gradle is invoked with --write-locks;"
+                            + " please remove 'org.gradle.configureondemand' from your gradle.properties");
```

Comment:
maybe pull this out to a package-private function?

Reply:
ah nvm the message is different

Reply:
🤷 i'd be happy to do that and modify the message thusly:
```
configure-on-demand cannot be enabled when gradle is invoked with '--write-locks' or when running the 'verifyLocks' task
```
if you prefer?

Reply:
up to you, i'm fine either way. shall we add a changelog?

Reply:
@dansanduleac doesn't look like i have permissions to add via changelog-bot? i can't click the button, apparently...

Reply:
I don't think it knows how to push to a fork unfortunately. I thought we gave everyone basic perms to push to these repos.

Might be easiest to write the changelog manually, like a cave-person.

Reply:
ah doh... ok

Reply:
alrighty, added manually